In [1]:
import numpy as np
import pandas as pd

import warnings
warnings.filterwarnings('ignore')

In [2]:
data1 = pd.read_excel('../DataSets/Principal_Commodity_Wise_Export/2014-2015.xls')
data2 = pd.read_excel('../DataSets/Principal_Commodity_Wise_Export/2015-2016.xls')
data3 = pd.read_excel('../DataSets/Principal_Commodity_Wise_Export/2016-2017.xls')
data4 = pd.read_excel('../DataSets/Principal_Commodity_Wise_Export/2017-2018.xls')

WARNING *** OLE2 inconsistency: SSCS size is 0 but SSAT size is non-zero
WARNING *** OLE2 inconsistency: SSCS size is 0 but SSAT size is non-zero
WARNING *** OLE2 inconsistency: SSCS size is 0 but SSAT size is non-zero
WARNING *** OLE2 inconsistency: SSCS size is 0 but SSAT size is non-zero


In [3]:
# Checking head for 2014-2015
data1.head(10)

PC_CODE     PC_DESCRIPTION UNIT  CTY_CODE      CTY_DESC  Qty  \
0      99  OTHER COMMODITIES  NaN       599   UNSPECIFIED  NaN   
1      99  OTHER COMMODITIES  NaN      1213         KENYA  NaN   
2      99  OTHER COMMODITIES  NaN      1395  TANZANIA REP  NaN   
3      99  OTHER COMMODITIES  NaN      1417        UGANDA  NaN   
4      99  OTHER COMMODITIES  NaN      2035         BENIN  NaN   
5      99  OTHER COMMODITIES  NaN      2050  BURKINA FASO  NaN   
6      99  OTHER COMMODITIES  NaN      2057      CAMEROON  NaN   
7      99  OTHER COMMODITIES  NaN      2067    C AFRI REP  NaN   
8      99  OTHER COMMODITIES  NaN      2069          CHAD  NaN   
9      99  OTHER COMMODITIES  NaN      2087   CONGO P REP  NaN   

   VAL(USD Million)  
0        371.280801  
1         18.251164  
2         21.844100  
3          8.034605  
4          1.812666  
5          0.388851  
6          0.893831  
7          0.520453  
8          0.008746  
9          1.305564

In [4]:
# Checking head for 2015-2016
data2.head(10)

PC_CODE     PC_DESCRIPTION UNIT  CTY_CODE      CTY_DESC  Qty  \
0      99  OTHER COMMODITIES  NaN       599   UNSPECIFIED  NaN   
1      99  OTHER COMMODITIES  NaN      1213         KENYA  NaN   
2      99  OTHER COMMODITIES  NaN      1395  TANZANIA REP  NaN   
3      99  OTHER COMMODITIES  NaN      1417        UGANDA  NaN   
4      99  OTHER COMMODITIES  NaN      2035         BENIN  NaN   
5      99  OTHER COMMODITIES  NaN      2050  BURKINA FASO  NaN   
6      99  OTHER COMMODITIES  NaN      2057      CAMEROON  NaN   
7      99  OTHER COMMODITIES  NaN      2067    C AFRI REP  NaN   
8      99  OTHER COMMODITIES  NaN      2069          CHAD  NaN   
9      99  OTHER COMMODITIES  NaN      2087   CONGO P REP  NaN   

   VAL(USD Million)  
0        159.558366  
1         16.088382  
2         13.965870  
3          9.371726  
4          1.564994  
5          0.465861  
6          0.465980  
7          0.173326  
8          0.141277  
9          1.564975

In [5]:
# Checking head for 2016-2017
data3.head(10)

PC_CODE     PC_DESCRIPTION UNIT  CTY_CODE      CTY_DESC  Qty  \
0      99  OTHER COMMODITIES  NaN       599   UNSPECIFIED  NaN   
1      99  OTHER COMMODITIES  NaN      1213         KENYA  NaN   
2      99  OTHER COMMODITIES  NaN      1395  TANZANIA REP  NaN   
3      99  OTHER COMMODITIES  NaN      1417        UGANDA  NaN   
4      99  OTHER COMMODITIES  NaN      2035         BENIN  NaN   
5      99  OTHER COMMODITIES  NaN      2050  BURKINA FASO  NaN   
6      99  OTHER COMMODITIES  NaN      2057      CAMEROON  NaN   
7      99  OTHER COMMODITIES  NaN      2067    C AFRI REP  NaN   
8      99  OTHER COMMODITIES  NaN      2069          CHAD  NaN   
9      99  OTHER COMMODITIES  NaN      2087   CONGO P REP  NaN   

   VAL(USD Million)  
0        270.889542  
1         17.159452  
2          8.896539  
3          5.181412  
4          2.476015  
5          0.452000  
6          0.448168  
7          0.001924  
8          0.004635  
9          1.747451

In [6]:
# Checking head for 2017-2018
data4.head(10)

COMMODITY UNIT_CODE      COUNTRY        QTY  VALUES (US$ million)
0       TEA       KGS  AFGHANISTAN   996994.0                  2.88
1       TEA       KGS      ALBANIA   250065.0                  0.82
2       TEA       KGS      ALGERIA   109640.0                  0.21
3       TEA       KGS      ANDORRA    32692.0                  0.15
4       TEA       KGS       ANGOLA    11361.0                  0.07
5       TEA       KGS    ARGENTINA       11.0                  0.00
6       TEA       KGS      ARMENIA     2311.0                  0.00
7       TEA       KGS    AUSTRALIA  3107210.0                 17.25
8       TEA       KGS      AUSTRIA     1164.0                  0.01
9       TEA       KGS   AZERBAIJAN   367209.0                  1.24

In [7]:
# Dropping PC_CODE and CTY_CODE column from first 3 tables
data1.drop(labels = ['PC_CODE', 'CTY_CODE'], axis = 1, inplace = True)
data2.drop(labels = ['PC_CODE', 'CTY_CODE'], axis = 1, inplace = True)
data3.drop(labels = ['PC_CODE', 'CTY_CODE'], axis = 1, inplace = True)

In [8]:
# Changing column names for first 3 tables to match column names in the fourth table
cols = data4.columns
data1.columns, data2.columns, data3.columns = cols, cols, cols

In [9]:
# Understanding the type of data in columns
data1[['COMMODITY', 'UNIT_CODE', 'COUNTRY']].describe()

COMMODITY UNIT_CODE COUNTRY
count               20830      9924   20830
unique                168         5     233
top     OTHER COMMODITIES       KGS   U S A
freq                  208      5381     160

From above table, we can see that 'COMMODITY', 'UNIT_TYPE' and 'COUNTRY' are categorical columns

In [10]:
data1[['QTY', 'VALUES (US$ million)']].describe()

QTY  VALUES (US$ million)
count  9.924000e+03          20830.000000
mean   3.731122e+06             14.899280
std    6.518345e+07            149.314062
min    0.000000e+00              0.000000
25%    1.540000e+02              0.028031
50%    5.658500e+03              0.324017
75%    1.708278e+05              2.969877
max    4.267870e+09           9439.054018

From above table, we can see that 'QTY' is Discrete(Numerical) and 'VALUES (US$ million)' is Continous(Numerical) column

In [11]:
# Checking null values in all the tables
data1.isnull().sum()

COMMODITY                   0
UNIT_CODE               10906
COUNTRY                     0
QTY                     10906
VALUES (US$ million)        0
dtype: int64

In [12]:
data2.isnull().sum()

COMMODITY                   0
UNIT_CODE               10779
COUNTRY                     0
QTY                     10779
VALUES (US$ million)        0
dtype: int64

In [13]:
data3.isnull().sum()

COMMODITY                   0
UNIT_CODE               10779
COUNTRY                     0
QTY                     10779
VALUES (US$ million)        0
dtype: int64

For first 3 tables, UNIT_CODE and QTY columns have null values. It means that for a commodity with unknown units the quantity exported is also unknown, which kinda make sense.

In [14]:
data4.isnull().sum()

COMMODITY                   0
UNIT_CODE                   1
COUNTRY                     1
QTY                     11478
VALUES (US$ million)        0
dtype: int64

In the fourth table UNIT_CODE has only 1 null value while QTY has 11478 null values. This is different from the pattern we saw in first 3 tables. It is important to visually inspect or check the value counts for columns in order to understand which value is null but python is interpreting it as a string.  

In [15]:
data4['UNIT_CODE'].value_counts()

N .A    11477
KGS      5394
TON      3818
NOS       484
SQM       295
LTR       112
Name: UNIT_CODE, dtype: int64

As expected, there is a value 'N .A' which is null but python is intrepreting it as a string and so is is not comming in the null count. We will replace this value with null.

In [16]:
# Replacing N .A with null in UNIT_CODE column for fourth table
data4['UNIT_CODE'].replace(to_replace = data4['UNIT_CODE'].value_counts().index[0], value = np.nan, inplace = True)

In [17]:
# Checking null value in fourth table
data4.isnull().sum()

COMMODITY                   0
UNIT_CODE               11478
COUNTRY                     1
QTY                     11478
VALUES (US$ million)        0
dtype: int64

We can still see a null value in COUNTRY column. So let's see the row for which COUNTRY column has null value. 

In [18]:
data4[data4['COUNTRY'].isnull()]

COMMODITY UNIT_CODE COUNTRY  QTY  VALUES (US$ million)
21580     Total       NaN     NaN  NaN             303523.19

There is a row which shows the total revenue. We don't need it and so we drop it. 

In [19]:
data4.drop(labels = 21580, axis = 0, inplace = True)

In [20]:
# Changing QTY from KGS to TON, dividing all quantities having unit KGS by 1000 to convert to TON in QTY column.
data1['QTY'] = data1[['UNIT_CODE', 'QTY']].apply(lambda x : x[1]/1000 if x[0] == 'KGS' else x[1], axis = 1)
data2['QTY'] = data2[['UNIT_CODE', 'QTY']].apply(lambda x : x[1]/1000 if x[0] == 'KGS' else x[1], axis = 1)
data3['QTY'] = data3[['UNIT_CODE', 'QTY']].apply(lambda x : x[1]/1000 if x[0] == 'KGS' else x[1], axis = 1)
data4['QTY'] = data4[['UNIT_CODE', 'QTY']].apply(lambda x : x[1]/1000 if x[0] == 'KGS' else x[1], axis = 1)

# Changing KGS to TON in UNIT_CODE column
data1['UNIT_CODE'].replace(to_replace = 'KGS', value = 'TON', inplace = True)
data2['UNIT_CODE'].replace(to_replace = 'KGS', value = 'TON', inplace = True)
data3['UNIT_CODE'].replace(to_replace = 'KGS', value = 'TON', inplace = True)
data4['UNIT_CODE'].replace(to_replace = 'KGS', value = 'TON', inplace = True)

In [21]:
data1['UNIT_CODE'].value_counts()

TON    9052
NOS     467
SQM     294
LTR     111
Name: UNIT_CODE, dtype: int64

<h5> Now all 4 dataframes are cleaned and are just having null values in UNIT_CODE and QTY columns. </h5> 

In [22]:
# Exporting all four dataframes into csv files
data1.to_csv('../DataSets/Principal_Commodity_Wise_Export/2014-2015.csv', index = False)
data2.to_csv('../DataSets/Principal_Commodity_Wise_Export/2015-2016.csv', index = False)
data3.to_csv('../DataSets/Principal_Commodity_Wise_Export/2016-2017.csv', index = False)
data4.to_csv('../DataSets/Principal_Commodity_Wise_Export/2017-2018.csv', index = False)